File: src/main/java/org/pyjjs/scheduler/core/api/Plan.java
package org.pyjjs.scheduler.core.api;

import org.pyjjs.scheduler.core.model.DateRange;
import org.pyjjs.scheduler.core.model.TaskResult;

import java.util.Set;

public interface Plan {

    DateRange getRange();

    Set<TaskResult> getPlan();
}

File: src/main/java/org/pyjjs/scheduler/core/api/Plan.kt
package org.pyjjs.scheduler.core.api

import org.pyjjs.scheduler.core.model.ResourceUsage
import java.util.*

interface Plan {
    var lastUpdate: Long
    var version: Long
    var differentWithIdeal: Double
    val resourceUsages: SortedSet<ResourceUsage>
}
